src/EditTableDialog.cpp

```diff
@@ -754,6 +765,10 @@ void EditTableDialog::constraintItemChanged(QTableWidgetItem* item)
     case kConstraintName:
         constraint->setName(item->text().toStdString());
         break;
+    case kConstraintReference:
+        auto fk = std::dynamic_pointer_cast<sqlb::ForeignKeyClause>(constraint);
```

Comment:
You need to check the constraint type here. For all constraints other than a foreign key (e.g. a check constraint), the ```dynamic_pointer_cast``` will return ```nullptr``` and the next line will crash.

To support other constraint types you might need to introduce a ```switch``` statement here:
```c++
switch(constraint->type())
{
case ForeignKeyConstraintType:
    auto fk = std::dynamic_pointer_cast<sqlb::ForeignKeyClause>(constraint);
    fk->setFromString(item->text().toStdString());
    break;
case CheckConstraintType:
    auto check = std::dynamic_pointer_cast<sqlb::CheckConstraint>(constraint);
    check->setExpression(item->text().toStdString());
    break;

// Add the other types here

}
```